mteb/models/mctct_model.py

```diff
@@ -0,0 +1,175 @@
+
+        # Convert to mono if needed
+        if waveform.shape[0] > 1:  # If multi-channel
+            waveform = torch.mean(waveform, dim=0, keepdim=True)  # Convert to mono
+
+        if sample_rate != self.sampling_rate:
+            resampler = torchaudio.transforms.Resample(sample_rate, self.sampling_rate)
+            waveform = resampler(waveform)
+        return waveform.squeeze()
+
+    def get_audio_embeddings(
+        self,
+        audio: AudioBatch,
+        *,
+        task_name: str | None = None,
+        prompt_type: PromptType | None = None,
+        batch_size: int = 4,
+        **kwargs: Any,
+    ) -> torch.Tensor:
+        processed_audio = self._process_audio(audio)
+        all_embeddings = []
+
+        with torch.no_grad():
+            for i in tqdm(range(0, len(processed_audio), batch_size)):
+                batch = processed_audio[i : i + batch_size]
+
+                # Process each audio in the batch
+                inputs = self.feature_extractor(
+                    [audio.cpu().numpy() for audio in batch],
+                    sampling_rate=self.sampling_rate,
+                    return_tensors="pt",
+                    padding=True,
+                ).to(self.device)
+
+                # Get embeddings from the model
+                outputs = self.model(
+                    input_features=inputs.input_features,
+                    attention_mask=inputs.attention_mask,
+                    output_hidden_states=True,
+                    return_dict=True,
+                )
+
+                # Get embeddings from the final layer hidden states
+                last_hidden = outputs.hidden_states[-1]
+                emb = last_hidden.mean(dim=1).cpu()
+                all_embeddings.append(emb)
+
+        return torch.cat(all_embeddings, dim=0)
+
+    def encode(
+        self,
+        inputs: AudioBatch,
+        *,
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        return self.get_audio_embeddings(inputs, task_name=task_name, **kwargs).numpy()
+
+
+mctct_large = ModelMeta(
+    loader=partial(
+        MCTCTWrapper,
+        model_name="speechbrain/m-ctc-t-large",
+    ),
+    name="speechbrain/m-ctc-t-large",
+    languages=["eng-Latn"],  # Supports 60 languages found in reference
+    open_weights=True,
+    revision="ed014c8255cea2c36f87a71cf2533b665ba00863",
+    release_date="2022-01-10",
+    max_tokens=None,
+    n_parameters=1_058_978_691,
+    memory_usage_mb=4039,
+    embed_dim=1536,  # Hidden dimension
+    license="apache-2.0",
+    reference="https://huggingface.co/speechbrain/m-ctc-t-large",
+    similarity_fn_name="cosine",
+    framework=["PyTorch"],
+    use_instructions=False,
+    public_training_code="https://github.com/speechbrain/speechbrain",
+    public_training_data="https://github.com/speechbrain/speechbrain",
+    training_datasets={},  # {"Common Voice": ["train"], "VoxPopuli": ["train"]},
```

Comment:
Maybe specify them explicitly?
```suggestion
    training_datasets={"CommonVoice": ["train"], "VoxPopuli": ["train"]},
```

Reply:
I thought it would cause problems if the datasets were not in MTEB, but I did the change in the latest commit.